bitbots_quintic_walk/include/bitbots_quintic_walk/walk_engine.h

```diff
@@ -28,14 +28,21 @@ namespace bitbots_quintic_walk {
  */
 class WalkEngine : public bitbots_splines::AbstractEngine<WalkRequest, WalkResponse> {
  public:
-  WalkEngine();
+  WalkEngine(const std::string ns);
 
   // methods from abstract engine class
   WalkResponse update(double dt) override;
   void setGoals(const WalkRequest &goals) override;
   void reset() override;
   int getPercentDone() const override;
 
+  void specialReset(WalkState state,
```

Comment:
I think a method docstring would be good given the "special" name of this method.

Reply:
Also I think you could just call this `reset` and overload the method as you did with `WalkNode::step`.